scalableminds/webknossos
# Show a warning before leaving the tracing view

When leaving the tracing view without having saved there should be warning. Currently there is only a warning when moving to page within the ReactRouter's control. (e.g. moving to dashboard) When reloading the page, such a message is not shown.

This PR adds a warning using the `beforeLeave` event to prevent accidental navigations when reloading.

### Steps to test:
- Trace some. While the saving indicator is still spinning reload the page. --> Warning
- Trace some. Click on dashboard or Admin.Users. --> Warning (different text)
- Trace some. Click save. Navigate away using either method --> no warning
- Open the same tracing in two tabs. Modify both and save them. --> A warning should appear on the page should be reloaded.

------
